client/verta/verta/operations/monitoring/alert/_entities/alert.py

```diff
@@ -0,0 +1,274 @@
+            alert=_AlertService.Alert(
+                name=name,
+                monitored_entity_id=monitored_entity_id,
+                created_at_millis=created_at_millis,
+                updated_at_millis=updated_at_millis,
+                last_evaluated_at_millis=last_evaluated_at_millis,
+                notification_channels={
+                    notification_channel.id: True
+                    for notification_channel in notification_channels
+                },
+                sample_find_base=summary_sample_query._to_proto_request(),
+                alerter_type=alerter._TYPE,
+            ),
+        )
+        if msg.alert.alerter_type == _AlertService.AlerterTypeEnum.FIXED:
+            msg.alert.alerter_fixed.CopyFrom(alerter._as_proto())
+        elif msg.alert.alerter_type == _AlertService.AlerterTypeEnum.REFERENCE:
+            msg.alert.alerter_reference.CopyFrom(alerter._as_proto())
+        else:
+            raise ValueError(
+                "unrecognized alert type enum value {}".format(msg.alert.alerter_type)
+            )
+
+        endpoint = "/api/v1/alerts/createAlert"
+        response = conn.make_proto_request("POST", endpoint, body=msg)
+        alert_msg = conn.must_proto_response(response, _AlertService.Alert)
+        return alert_msg
+
+    def _update(self):
+        raise NotImplementedError
+
+    def add_notification_channel(self, notification_channel):
+        raise NotImplementedError
+
+    # TODO: alternatively, fire() & resolve()?
+    def set_status(self, status, summary_sample=None, event_time_millis=None):
+        msg = _AlertService.UpdateAlertStatusRequest(
+            alert_id=self.id,
+            event_time_millis=event_time_millis,
+            status=status._ALERT_STATUS,
+            violating_summary_sample_id=summary_sample.id,
+        )
+        endpoint = "/api/v1/alerts/updateAlertStatus"
+        response = self._conn.make_proto_request("POST", endpoint, body=msg)
+        self._conn.must_response(response)
+        return True
+
+    def delete(self):
+        msg = _AlertService.DeleteAlertRequest(ids=[self.id])
+        endpoint = "/api/v1/alerts/deleteAlert"
+        response = self._conn.make_proto_request("DELETE", endpoint, body=msg)
+        self._conn.must_response(response)
+        return True
+
+
+class Alerts(object):
+    def __init__(self, conn, conf, monitored_entity_id):
+        self._conn = conn
+        self._conf = conf
+        self._monitored_entity_id = int(monitored_entity_id)
+
+    def create(
+        self,
+        name,
+        alerter,
+        summary_sample_query,
+        notification_channels=None,
+        created_at_millis=None,
+        updated_at_millis=None,
+        last_evaluated_at_millis=None,
+    ):
+        if notification_channels is None:
+            notification_channels = []
+
+        for channel in notification_channels:
+            if isinstance(channel, notification_channel._NotificationChannel):
```

Comment:
This looks funny. Should it be `if not isinstance`?

Reply:
I guess this is as confusing as I feared it'd be 😕 

So, there are these config structs that are used to specify different types of notification channels, similar to how the data type classes work
```python
channel_config = SlackNotificationChannel(
    "https://hooks.slack.com/...",
)
```

And then there's the actual entity that exists in our platform
```python
channel_entity = client.notification_channels.create(
    "test-webhook",
    channel_config,
)
```

So what this check here does is let the user know that when creating an alert, they'd need to pass in the entity, not the config struct. I can add a comment explaining this `if` condition, though this may be a deeper issue with the API 😓

Reply:
Sure, but wouldn't we still want to test against the expected/supported type and raise an error if we get something else? For example, if we had a list of animals which really needed to all be dogs, it would look funny to raise an error if any of them were cats because then an error wouldn't be raised on elements which are birds.